Month 9 step 1

Konstantin0099 opened the pull request.

Konstantin0099 committed:
month-9_step-1>>

Konstantin0099 committed:
добавлен роут /login

Konstantin0099 committed:
добавлен роут /regiter

Konstantin0099 committed:
добавлен роут /profile

Konstantin0099 committed:
добавлен роут /reset-password

Konstantin0099 committed:
feat: добавить руты  forgot-password,reset-password,ingredients/:id, 404, order-feeed, order-history

Konstantin0099 committed:
feat: настройка форм

Konstantin0099 committed:
feat: проверка доступности маршрутов

Konstantin0099 committed:
fix: чистка кода

gennady-bars commented on src/services/thunk/auth-user.js:
```diff
@@ -0,0 +1,47 @@
+      })
+      .catch((e) => {
+        console.log("упс... ошибка function userAuthLogin :(", e);
+        dispatch({ type: AUTH_FAILED });
```
**Можно лучше**

`        dispatch({ type: AUTH_FAILED });
`


Чтобы не дублировать типы экшенов и сами объекты с `type`, принято выносить их в отдельные функции, называемые `action creator` (функцию, которая возвращает объект с `type`). Это позволяет сделать понятное название для функции, которое будет говорить, что делает она, и избавит от повтора объекта с `type` в коде. Подробнее тут: [https://rajdee.gitbooks.io/redux-in-russian/content/docs/basics/Actions.html#генераторы-экшенов-action-creators](https://rajdee.gitbooks.io/redux-in-russian/content/docs/basics/Actions.html#%D0%B3%D0%B5%D0%BD%D0%B5%D1%80%D0%B0%D1%82%D0%BE%D1%80%D1%8B-%D1%8D%D0%BA%D1%88%D0%B5%D0%BD%D0%BE%D0%B2-action-creators)

gennady-bars commented on src/pages/forgot-password.js:
```diff
@@ -0,0 +1,46 @@
+      </h2>
+      <ul className={style.list}>
+        <li className={style.field + " mb-4"}>
+          <EMail setData={setData} />
```
Если есть инпуты, то они должны быть внутри тега `form`, и на этот тег `form` нужно навешивать обработчик `onSubmit`, а не обработчик `onClick` на кнопку сабмита.

gennady-bars commented on src/pages/forgot-password.js:
```diff
@@ -0,0 +1,46 @@
+          <EMail setData={setData} />
+        </li>
+      </ul>
+      <Button type="primary" size="medium" onClick={recreate}>
```
`onClick={recreate}`

это нужно удалить с кнопки


Обработчик сабмита нужно навешивать только на тег `form` с событием `submit`, а не на кнопку сабмита с событием `click`, так как сабмит формы происходит ещё при нажатии `Enter`, и он не будет работать, если навесить обработчик клика на кнопку только. Это нужно исправить везде, где есть инпуты и форма

gennady-bars commented on src/pages/forgot-password.js:
```diff
@@ -0,0 +1,46 @@
+  const recreate = () => {
+    dispatch(forgotPassword(history, email));
+  };
+  React.useEffect(() => {}, [history]);
```
это пустой эффект. Нужно удалить

gennady-bars commented on src/components/protected-route/protected-route.js:
```diff
@@ -0,0 +1,32 @@
+    <Route
+      {...rest}
+      render={() =>
+        localStorage.getItem("refreshToken") ? (
```
наличие токена в `localStorage` не означает, что пользователя можно пускать в защищенный рут `ProtectedRoute`. Нужно 1 раз при автологине проверить токен и потом сделать флаг `isLoggedIn` и брать его из редакса. И по нему уже определять, залогинен ли пользователь сейчас.

gennady-bars commented on src/pages/forgot-password.js:
```diff
@@ -0,0 +1,46 @@
+    setEmail(email);
+  };
+
+  return localStorage.getItem("refreshToken") ? (
```
наличие токена в `localStorage` не означает, что пользователя можно пускать в защищенный рут `ProtectedRoute`. Нужно 1 раз при автологине проверить токен и потом сделать флаг `isLoggedIn` и брать его из редакса. И по нему уже определять, залогинен ли пользователь сейчас.

gennady-bars commented on src/pages/register.js:
```diff
@@ -0,0 +1,65 @@
+  const setData = (data, name) => {
+    setNewData({...newData, [name]: data});
+  }
+  return localStorage.getItem("refreshToken") ? (
```
наличие токена в `localStorage` не означает, что пользователя можно пускать в защищенный рут `ProtectedRoute`. Нужно 1 раз при автологине проверить токен и потом сделать флаг `isLoggedIn` и брать его из редакса. И по нему уже определять, залогинен ли пользователь сейчас.

gennady-bars commented on src/components/protected-route/protected-route.js:
```diff
@@ -0,0 +1,32 @@
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  useEffect(() => {}, []);
```
`  useEffect(() => {}, []);
`

пустые эффекты нужно удалить

gennady-bars commented on src/pages/login.js:
```diff
@@ -0,0 +1,64 @@
+    history.replace({ state: { revert: "/" } });
+    dispatch(userAuthLogin(history, newData, location.state.revert));
+  };
+  useEffect(() => {}, []);
```
`  useEffect(() => {}, []);
`

пустые эффекты нужно удалить

gennady-bars commented on src/components/app/app.js:
```diff
@@ -1,33 +1,99 @@
+  const { visible, user, data } = useSelector((state) => state);
   const dispatch = useDispatch();
+  const history = useHistory();
+  React.useEffect(() => {
```
пустой эффект нужно удалить

gennady-bars commented on src/components/app/app.js:
```diff
@@ -1,33 +1,99 @@
+          ></Modal>
+        )}
+        <DndProvider backend={HTML5Backend}>
+          <Switch>
```
Вам нужно использовать средства роутинга для определения, показывать ли попап с деталями, или показывать страницу. Вам должны были дать ссылку на документацию по этой логике. Вот она https://reactrouter.com/web/example/modal-gallery . Пожалуйста, посоветуйтесь с наставником.

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная, но есть некоторые недочеты:



* под попапом ингредиента виднеется страница ингредиента, а такого не должно быть. Скрин http://joxi.ru/D2PnJ9QCwwWNpr . Страница появляться должна только, если вставили ссылку на детали в браузер (или перезагрузили страницу)
* Если есть инпуты, то они должны быть внутри тега `form`, и на этот тег `form` нужно навешивать обработчик `onSubmit`, а не обработчик `onClick` на кнопку сабмита.
* Обработчик сабмита нужно навешивать только на тег `form` с событием `submit`, а не на кнопку сабмита с событием `click`, так как сабмит формы происходит ещё при нажатии `Enter`, и он не будет работать, если навесить обработчик клика на кнопку только. Это нужно исправить везде, где есть инпуты и форма 
* наличие токена в `localStorage` не означает, что пользователя можно пускать в защищенный рут `ProtectedRoute`. Нужно 1 раз при автологине проверить токен и потом сделать флаг `isLoggedIn` и брать его из редакса. И по нему уже определять, залогинен ли пользователь сейчас.
* Вам нужно использовать средства роутинга для определения, показывать ли попап с деталями, или показывать страницу. Вам должны были дать ссылку на документацию по этой логике. Вот она https://reactrouter.com/web/example/modal-gallery . Пожалуйста, посоветуйтесь с наставником.


**Можно лучше**

*  желательно очищать конструктор после успешного получения номера заказа с сервера в блоке `then`, чтобы пользователь мог следующий заказ сделать, не удаляя старые ингредиенты
*  при клике на логотип обычно идет переход на главную страницу
* Чтобы не дублировать типы экшенов и сами объекты с `type`, принято выносить их в отдельные функции, называемые `action creator` (функцию, которая возвращает объект с `type`). Это позволяет сделать понятное название для функции, которое будет говорить, что делает она, и избавит от повтора объекта с `type` в коде. Подробнее тут: [https://rajdee.gitbooks.io/redux-in-russian/content/docs/basics/Actions.html#генераторы-экшенов-action-creators](https://rajdee.gitbooks.io/redux-in-russian/content/docs/basics/Actions.html#%D0%B3%D0%B5%D0%BD%D0%B5%D1%80%D0%B0%D1%82%D0%BE%D1%80%D1%8B-%D1%8D%D0%BA%D1%88%D0%B5%D0%BD%D0%BE%D0%B2-action-creators)


Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

Konstantin0099 committed:
feat: верстка страниц /feed, /feed/id

Konstantin0099 committed:
feat: добавить socketMiddleware, wsReducers, wsActionTypes

Konstantin0099 committed:
feat: получить данные с WS отрисовать историю заказов

Konstantin0099 committed:
feat:
откорректировать работу модального окна
импуты завернуть в form
регистрацию пользователя проверять по значению успешной регистрации в state
использовать средства роутинга для отображения модального окна

Konstantin0099 committed:
feat: чистка кода

Konstantin0099 committed:
feat: добавить ренедер feed/:id

gennady-bars commented on src/pages/login.js:
```diff
@@ -17,21 +17,25 @@ import { logout } from "../services/thunk/logout";
+  return authSuccess ? <Redirect to="/" /> : (
     <div className={style.modal}>
       <h2 className={"text text_type_main-medium " + style.title}> ВХОД </h2>
+      <form onsubmit={getUser}>
```
* Вы неправильно указали атрибут `onSubmit` в тегах формы, поэтому сабмит не работает. Скрин http://joxi.ru/Q2KZl97HOgXlVm . Вам нужно удалить `onClick` со всех кнопок в формах

gennady-bars commented on src/components/profile-form/profile-form.js:
```diff
@@ -1,84 +1,72 @@
-      </div>
-      }
-    </ul>
+    <form onsubmit={setUser} className={style.form}>
```
* Вы неправильно указали атрибут `onSubmit` в тегах формы, поэтому сабмит не работает. Скрин http://joxi.ru/Q2KZl97HOgXlVm . Вам нужно удалить `onClick` со всех кнопок в формах

gennady-bars commented on src/pages/forgot-password.js:
```diff
@@ -9,25 +9,27 @@ import { EMail } from "../components/email/email";
         {" "}
         Востановление пароля{" "}
       </h2>
+      <form onsubmit={recreate}>
```
* Вы неправильно указали атрибут `onSubmit` в тегах формы, поэтому сабмит не работает. Скрин http://joxi.ru/Q2KZl97HOgXlVm . Вам нужно удалить `onClick` со всех кнопок в формах

gennady-bars commented on src/pages/reset-password.js:
```diff
@@ -31,6 +31,7 @@ export const ResetPassword = () => {
         {" "}
         Востановление пароля{" "}
       </h2>
+      <form onsubmit={resetPass}>
```
* Вы неправильно указали атрибут `onSubmit` в тегах формы, поэтому сабмит не работает. Скрин http://joxi.ru/Q2KZl97HOgXlVm . Вам нужно удалить `onClick` со всех кнопок в формах

gennady-bars commented on src/pages/register.js:
```diff
@@ -22,14 +23,15 @@ export const RegisterPage = () => {
         {" "}
         Регистрация{" "}
       </h2>
+      <form onsubmit={register}>
```
* Вы неправильно указали атрибут `onSubmit` в тегах формы, поэтому сабмит не работает. Скрин http://joxi.ru/Q2KZl97HOgXlVm . Вам нужно удалить `onClick` со всех кнопок в формах

gennady-bars reviewed (commented):
К сожалению, работу придется отклонить без потери итерации

* приложение ломается при старте. Скрин http://joxi.ru/823Y6BpSzwKxnr 
* Вы неправильно указали атрибут `onSubmit` в тегах формы, поэтому сабмит не работает. Скрин http://joxi.ru/Q2KZl97HOgXlVm . Вам нужно удалить `onClick` со всех кнопок в формах

Konstantin0099 committed:
fix: резервная копия

Konstantin0099 committed:
fix: чистка кода

Konstantin0099 committed:
fix: добавить чистый код из степ 2Э

Konstantin0099 committed:
fix: объединить ветки

Konstantin0099 committed:
fix: исправить ошибки

gennady-bars reviewed (commented):
К сожалению, работу придется отклонить без потери итерации, так как проект не запускается. Скрин ошибки http://joxi.ru/KAxkxl9SVwJ4v2 


* Вам не нужно было делать вебсокеты в этой же ветке. Они должны быть в другой. Сейчас намешается код, а потом невозможно будет его проверить в следующей итерации, так как не будет измененных файлов в той ветке, которую на ревью отправите

Konstantin0099 committed:
fix: проверка фиксации

Konstantin0099 committed:
fix: убрать запрос к websocket , до открытия необходимой страницы

Konstantin0099 committed:
fix: убрать console

Konstantin0099 committed:
fix: изменить проверку открыта лента или история

Konstantin0099 committed:
fix: убрать ошибки при отображении деталей заказа

gennady-bars commented on src/pages/forgot-password.js:
```diff
@@ -0,0 +1,46 @@
+            <EMail setData={setData} />
+          </li>
+        </ul>
+        <Button type="primary" size="medium" onSubmit={recreate}>
```
`onSubmit={recreate}` нужно удалить с кнопок сабмита. Обработка сабмита вешается только на тег `form`

gennady-bars commented on src/components/protected-route/protected-route.js:
```diff
@@ -0,0 +1,20 @@
+export function ProtectedRoute({ children, ...rest }) {
+  const { authSuccess } = useSelector((state) => state.user);
+  const history = useHistory();
+  React.useEffect(() => {}, []);
```
`  React.useEffect(() => {}, []);
`

пустые эффекты нужно удалить. Это лишняя нагрузка на реакт. Он их записывает в память

Konstantin0099 committed:
fix: чистка кода

gennady-bars reviewed (commented):
Еще немного и все будет отлично:

* `onSubmit={...}` нужно удалить с кнопок сабмита. Обработка сабмита вешается только на тег `form`
* пустые эффекты нужно удалить. Это лишняя нагрузка на реакт. Он их записывает в память
* нет автологина. Я логинюсь, потом перезагружаю страницу и меня опять просят залогиниться




**Можно лучше**

* Вам нужно использовать средства роутинга для определения, показывать ли попап с деталями, или показывать страницу. Вам должны были дать ссылку на документацию по этой логике. Вот она https://reactrouter.com/web/example/modal-gallery .

Konstantin0099 committed:
fix: корректировка стилей

Konstantin0099 committed:
fix: корректировка стилей

Konstantin0099 committed:
fix: реализовать автологин

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
